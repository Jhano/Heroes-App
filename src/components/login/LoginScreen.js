import React,{useContext} from 'react';
import { Button, Col, Container, Row } from 'react-bootstrap';
import AuthContext from '../../auth/AuthContext';
import { types } from '../../types/types';


const LoginScreen = ({history}) => {

    const {user, dispatch} = useContext(AuthContext);

    const handleLogin = () => {
        //history.replace('/'); //reemplaza en el historial
       
        const lastPath = localStorage.getItem('lastPath') || '/';

        dispatch({
            type: types.login,
            payload:  {
                name: 'Alejandro'
            }
        }) 
        
        history.replace(lastPath);
    
    }

    return ( 
            <Container  className="mt-5" >
                <Row> 
                    <Col lg="12">
                        <h1 style={{color: 'white'}}>LoginScreen</h1>
                        
                            <pre style={{color: 'white'}} >{JSON.stringify(user, null, 3)}</pre>
                          
                        <hr/>  
                    </Col> 
                    <Col xs="4" md="4" lg="2">
                        <Button 
                            variant="primary"
                            onClick={handleLogin}
                            size="lg" block
                        >
                            Login
                        </Button>
                    </Col>   
                    
                </Row>       
            </Container>
        
       
        
     );
}
 
export default LoginScreen;

