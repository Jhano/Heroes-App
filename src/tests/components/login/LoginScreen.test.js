import React from 'react';
import { mount } from 'enzyme';
import LoginScreen from '../../../components/login/LoginScreen';
import AuthContext from '../../../auth/AuthContext';
import { types } from '../../../types/types';



describe('Pruebas en <LoginScreen />', () => {

    
    const historyMock = {
        replace: jest.fn(),      
    }

    const contextValue = {
        user: {
            logged: false
        },
        dispatch: jest.fn()
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
             <LoginScreen history={historyMock} />
        </AuthContext.Provider>
       
    )

    test('debe mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('debe de realizar el dispatch y la navegacion', () => {

        const handleCLick = wrapper.find('button').prop('onClick');

        handleCLick();
        expect(contextValue.dispatch).toHaveBeenCalledWith(
                        {
                            type: types.login,
                            payload:  {
                            name: 'Alejandro'
                            }
                        }
                    );
        
         expect(historyMock.replace).toHaveBeenCalledWith('/');

         localStorage.setItem('lastPath', "/dc")   
         handleCLick();    
         expect(historyMock.replace).toHaveBeenCalledWith('/dc');
                                        
    })
})