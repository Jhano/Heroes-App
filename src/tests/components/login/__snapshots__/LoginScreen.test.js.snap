// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> debe mostrarse correctamente 1`] = `
<div
  className="mt-5 container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <h1
        style={
          Object {
            "color": "white",
          }
        }
      >
        LoginScreen
      </h1>
      <pre
        style={
          Object {
            "color": "white",
          }
        }
      >
        {
   "logged": false
}
      </pre>
      <hr />
    </div>
    <div
      className="col-lg-2 col-md-4 col-4"
    >
      <button
        className="btn btn-primary btn-block btn-lg"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
